blockchain/tx_pool.go

```diff
@@ -1971,6 +1999,80 @@ func (pool *TxPool) RegisterTxPoolModule(modules ...kaiax.TxPoolModule) {
 	pool.modules = modules
 }
 
+func (pool *TxPool) sendMissingBlobSidecar(sidecar *MissingBlobSidecar) {
+	select {
+	case pool.missingBlobSidecarsCh <- sidecar:
+	default:
+		logger.Debug("Missing blob sidecars channel is full, dropping notification", "txHash", sidecar.TxHash)
+	}
+}
+
+func (pool *TxPool) SubscribeMissingBlobSidecars() <-chan *MissingBlobSidecar {
+	return pool.missingBlobSidecarsCh
+}
+
+func (pool *TxPool) SaveBlobSidecar(blockNum *big.Int, txIndex int, txHash common.Hash, sidecar *types.BlobTxSidecar) error {
+	if pool.blobStorage == nil || blockNum == nil {
+		return errors.New("blob storage not initialized")
+	}
+	if sidecar == nil {
+		return errors.New("sidecar is nil")
+	}
+	// TODO: verify sidecar
+	// tx, _, _, _ := pool.chain.GetTxAndLookupInfo(txHash)
+	// if  tx == nil {
+	// 	return errors.New("tx not found")
+	// }
+	// if err := sidecar.ValidateWithBlobTx(tx.GetTxInternalData().(*types.TxInternalDataEthereumBlob)); err != nil {
+	// 	return err
+	// }
+
+	return pool.blobStorage.Save(blockNum, txIndex, sidecar)
+}
+
+// saveAndPruneBlobStorage saves and prunes blob storage.
+func (pool *TxPool) saveAndPruneBlobStorage(newHead *types.Block) {
+	if pool.blobStorage == nil || newHead == nil {
+		return
+	}
+
+	// skip if the block is too old
+	if time.Since(time.Unix(int64(newHead.Time().Uint64()), 0)) > 3*7*24*time.Hour {
```

Comment:
Use pool.BlobStorageConfig.Retention

Reply:
fix at https://github.com/kaiachain/kaia/pull/681/changes/7f5f7b7fcacd340250e6a5e21a66348aca04815f

Reply:
<s>You're still using the hardcoded 3*7*24*time.Hour.</s> I was seeing an old commits.